pyteal/ast/app.py

```diff
@@ -356,3 +356,56 @@ def address(cls, app: Expr) -> MaybeValue:
 
 
 AppParam.__module__ = "pyteal"
+
+
+class AppParamObject:
+    """Represents information about an application's parameters"""
+
+    def __init__(self, app: Expr) -> None:
+        """Create a new AppParamObject for the given application.
+
+        Args:
+            app: An identifier for the app. It must be an index into Txn.ForeignApps that
+                corresponds to the app to check, or since v4, an application ID that appears in
+                Txn.ForeignApps or is the CurrentApplicationID. In either case, it must evaluate to
+                uint64.
+        """
+        self._app: Final = app
```

Comment:
nit: do we want to type check here, for `_app` must be evaluated to `uint64` by comment?